lib/components/Web3Context/index.tsx

```diff
@@ -10,10 +11,16 @@ interface Props {
 }
 /** Init the web3Modal and expose via react context  */
 export const Web3ContextProvider: FC<Props> = (props) => {
-  const web3ProviderState = useWeb3Modal();
-
+  const providerState = useProvider();
+  const [showModal, setShowModal] = useState(false);
+  const toggleModal = () => setShowModal((s: boolean) => !s);
+  const renderModal = (props: any) => (
```

Comment:
Provide a type for `props`
You can export the ConnectModalProps or slightly more ugly you can do `React.ComponentProps<typeof ConnectModal>`